### PR TITLE
2873 update logic in additional ein generator to use census models

### PR DESCRIPTION
**Pull request by sambodeme**
## Description
Updated the additional EIN/UEI workbook generators logic to use Census models from `models.py`, replacing the previously used custom models from `census.py`. This involved:

- Modifying references to field names to align with the Census models.
- Reorganizing and refactoring parts of the logic to enhance reusability.
- Transitioning from custom queries to standard Django ORM query styles.

## How to Test:

1. Start with a clean environment:

        - Run make docker-clean
        - Followed by make docker-first-run
        - Then execute docker compose up

2. Execute the steps described in the [how-to-load-test-census-into-postgres](https://github.com/GSA-TTS/FAC/blob/main/backend/census_historical_migration/README.md#how-to-load-test-census-data-into-postgres) of the README.
3. At this point, you can opt to test either the [historic workbook generator](https://github.com/GSA-TTS/FAC/blob/main/backend/census_historical_migration/README.md#how-to-run-the-historic-workbook-generator) or the [historic data migrator](https://github.com/GSA-TTS/FAC/blob/main/backend/census_historical_migration/README.md#how-to-run-the-historic-data-migrator). To perform these tests:

      - Use any dbkey found in the first part of the subfolder names located in `census_historical_migration/fixtures/workbooks/should_pass`. The subfolders are named using the pattern `dbkey-year`.
      - Do not use dbkey 69688 (that audit report requires further investigation)
## PR checklist: submitters

- [ ]   Link to an issue if possible. If there’s no issue, describe what your branch does. Even if there is an issue, a brief description in the PR is still useful.
- [ ]   List any special steps reviewers have to follow to test the PR. For example, adding a local environment variable, creating a local test file, etc.
- [ ]   For extra credit, submit a screen recording like [this one](https://github.com/GSA-TTS/FAC/pull/1821).
- [ ]   Make sure you’ve merged `main` into your branch shortly before creating the PR. (You should also be merging `main` into your branch regularly during development.)
- [ ]   Make sure you’ve accounted for any migrations. When you’re about to create the PR, bring up the application locally and then run `git status | grep migrations`. If there are any results, you probably need to add them to the branch for the PR. Your PR should have only **one** new migration file for each of the component apps, except in rare circumstances; you may need to delete some and re-run `python manage.py makemigrations` to reduce the number to one. (Also, unless in exceptional circumstances, your PR should not delete any migration files.)
- [ ]   Make sure that whatever feature you’re adding has tests that cover the feature. This includes test coverage to make sure that the previous workflow still works, if applicable.
- [ ]   Make sure the full-submission.cy.js [Cypress test](https://github.com/GSA-TTS/FAC/blob/main/docs/testing.md#end-to-end-testing) passes, if applicable.
- [ ]   Do manual testing locally. Our tests are not good enough yet to allow us to skip this step. If that’s not applicable for some reason, check this box.
- [ ]   Verify that no Git surgery was necessary, or, if it was necessary at any point, repeat the testing after it’s finished.
- [ ]   Once a PR is merged, keep an eye on it until it’s deployed to dev, and do enough testing on dev to verify that it deployed successfully, the feature works as expected, and the happy path for the broad feature area (such as submission) still works.

## PR checklist: reviewers

- [ ]   Pull the branch to your local environment and run `make docker-clean; make docker-first-run && docker compose up`; then run `docker compose exec web /bin/bash -c "python manage.py test"`
- [ ]   Manually test out the changes locally, or check this box to verify that it wasn’t applicable in this case.
- [ ]   Check that the PR has appropriate tests. Look out for changes in HTML/JS/JSON Schema logic that may need to be captured in Python tests even though the logic isn’t in Python.
- [ ]   Verify that no Git surgery is necessary at any point (such as during a merge party), or, if it was, repeat the testing after it’s finished.

The larger the PR, the stricter we should be about these points.

### CODE REVIEW

File: backend/census_historical_migration/sac_general_lib/report_id_generator.py [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
+    xform_census_date_to_datetime,
+)
 
 
 # FIXME: Get the padding/shape right on the report_id
-def dbkey_to_report_id(Gen, dbkey):
+def dbkey_to_report_id(
+    Gen, dbkey
+):  # FIXME-MSHD: Remove this function once we switch all workbook generators to using census models
     g = Gen.select(Gen.audityear, Gen.fyenddate).where(Gen.dbkey == dbkey).get()
```
Comment (JeanMarie-TTS) [inline backend/census_historical_migration/sac_general_lib/report_id_generator.py:11]: Can we switch to using the Django ORM please?

File: backend/census_historical_migration/workbooklib/additional_ueis.py [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
     generate_dissemination_test_table,
+    set_workbook_uei,
 )
 from census_historical_migration.base_field_maps import (
     SheetFieldMap,
     WorkbookFieldInDissem,
 )
 from census_historical_migration.workbooklib.templates import sections_to_template_paths
-from census_historical_migration.workbooklib.census_models.census import dynamic_import
+from census_historical_migration.models import ELECUEIS as Ueis
```
Comment (JeanMarie-TTS) [inline backend/census_historical_migration/workbooklib/additional_ueis.py:12]: We should use relative imports within the same app. i,e, `from .models import ELECUEIS as Ueis`

File: backend/census_historical_migration/workbooklib/excel_creation_utils.py [hunk 3]
```diff
@@ -1,11 +1,19 @@
+from census_historical_migration.transforms.xform_string_to_string import (
+    string_to_string,
+)
+from census_historical_migration.transforms.xform_string_to_int import string_to_int
```
Comment (JeanMarie-TTS) [inline backend/census_historical_migration/workbooklib/excel_creation_utils.py:4]: We should use relative imports within the same app

File: backend/census_historical_migration/workbooklib/findings_text.py [hunk 4]
```diff
@@ -19,29 +20,43 @@
...
-    Findingstext = dynamic_import("Findingstext", year)
+
     wb = pyxl.load_workbook(sections_to_template_paths[FORM_SECTIONS.FINDINGS_TEXT])
 
-    g = set_uei(Gen, wb, dbkey)
+    audit_header = get_audit_header(dbkey)
+
+    set_workbook_uei(wb, audit_header.UEI)
+
+    findings_texts = _get_findings_text(dbkey)
```
Comment (phildominguez-gsa) [inline backend/census_historical_migration/workbooklib/findings_text.py:51]: I would make `findings_texts` and `_get_findings_text()` either both singular or both plural

File: backend/census_historical_migration/workbooklib/secondary_auditors.py [hunk 5]
```diff
@@ -16,6 +16,19 @@
 
 logger = logging.getLogger(__name__)
 
+
+def xform_add_hyphen_to_zip(zip):
```
Comment (phildominguez-gsa) [inline backend/census_historical_migration/workbooklib/secondary_auditors.py:20]: Docstring would be good

File: backend/census_historical_migration/workbooklib/findings_text.py [hunk 6]
```diff
@@ -19,29 +20,43 @@
...
+
+    map_simple_columns(wb, mappings, findings_texts)
 
-    ftexts = Findingstext.select().where(Findingstext.dbkey == g.dbkey)
-    map_simple_columns(wb, mappings, ftexts)
     wb.save(outfile)
+
     table = generate_dissemination_test_table(
```
Comment (phildominguez-gsa) [inline backend/census_historical_migration/workbooklib/findings_text.py:57]: My recommendation for spacing so it's a bit easier to follow:

```
    wb = pyxl.load_workbook(sections_to_template_paths[FORM_SECTIONS.FINDINGS_TEXT])
    audit_header = get_audit_header(dbkey)
    set_workbook_uei(wb, audit_header.UEI)

    findings_texts = _get_findings_text(dbkey)
    map_simple_columns(wb, mappings, findings_texts)

    wb.save(outfile)
```

File: backend/census_historical_migration/workbooklib/additional_ueis.py [hunk 7]
```diff
@@ -19,26 +20,33 @@
...
-    else:
-        table = {}
-        table["singletons"] = {}
+    audit_header = get_audit_header(dbkey)
+
+    set_workbook_uei(wb, audit_header.UEI)
+
+    additional_ueis = _get_ueis(dbkey)
+    map_simple_columns(wb, mappings, additional_ueis)
     wb.save(outfile)
```
Comment (phildominguez-gsa) [inline backend/census_historical_migration/workbooklib/additional_ueis.py:44]: My recommendation for spacing so it's a bit easier to follow:

```
    wb = pyxl.load_workbook(sections_to_template_paths[FORM_SECTIONS.ADDITIONAL_UEIS])
    audit_header = get_audit_header(dbkey)
    set_workbook_uei(wb, audit_header.UEI)

    additional_ueis = _get_ueis(dbkey)
    map_simple_columns(wb, mappings, additional_ueis)

    wb.save(outfile)
```

File: backend/census_historical_migration/workbooklib/additional_eins.py [hunk 8]
```diff
@@ -17,25 +21,49 @@
...
 
-    g = set_uei(Gen, wb, dbkey)
+    set_workbook_uei(wb, audit_header.UEI)
 
-    addl_eins = Eins.select().where(Eins.dbkey == g.dbkey)
+    addl_eins = _get_eins(dbkey)
     map_simple_columns(wb, mappings, addl_eins)
     wb.save(outfile)
-
+    # FIXME - MSHD: The logic below will most likely be removed, see comment in federal_awards.py
```
Comment (phildominguez-gsa) [inline backend/census_historical_migration/workbooklib/additional_eins.py:64]: My recommendation for spacing so it's a bit easier to follow:
```
    wb = pyxl.load_workbook(sections_to_template_paths[FORM_SECTIONS.ADDITIONAL_EINS])
    audit_header = get_audit_header(dbkey)
    set_workbook_uei(wb, audit_header.UEI)

    addl_eins = _get_eins(dbkey)
    map_simple_columns(wb, mappings, addl_eins)

    wb.save(outfile)

    # FIXME - MSHD: The logic below will most likely be removed, see comment in federal_awards.py
```

**Review by phildominguez-gsa — CHANGES_REQUESTED** [review]

**Review by phildominguez-gsa — APPROVED** [review]

**Pull request merged**